hypothesis-python/src/hypothesis/extra/cli.py

```diff
@@ -84,27 +85,64 @@ def obj_name(s: str) -> object:
             return importlib.import_module(s)
         except ImportError:
             pass
+        classname = None
         if "." not in s:
             modulename, module, funcname = "builtins", builtins, s
         else:
             modulename, funcname = s.rsplit(".", 1)
             try:
                 module = importlib.import_module(modulename)
             except ImportError as err:
+                try:
+                    modulename, classname = modulename.rsplit(".", 1)
+                    module = importlib.import_module(modulename)
+                except ImportError:
+                    raise click.UsageError(
+                        f"Failed to import the {modulename} module for introspection.  "
+                        "Check spelling and your Python import path, or use the Python API?"
+                    ) from err
+        if classname is None:
+            try:
+                return getattr(module, funcname)
+            except AttributeError as err:
+                public_names = [
+                    name for name in vars(module) if not name.startswith("_")
+                ]
+                matches = get_close_matches(funcname, public_names)
                 raise click.UsageError(
-                    f"Failed to import the {modulename} module for introspection.  "
-                    "Check spelling and your Python import path, or use the Python API?"
+                    f"Found the {modulename!r} module, but it doesn't have a "
+                    f"{funcname!r} attribute."
+                    + (f"  Closest matches: {matches!r}" if matches else "")
+                ) from err
+        else:
+            try:
+                func_class = getattr(module, classname)
+            except AttributeError as err:
+                public_names = [
+                    name for name in vars(module) if not name.startswith("_")
+                ]
+                matches = get_close_matches(classname, public_names)
+                raise click.UsageError(
+                    f"Found the {modulename!r} module, but it doesn't have a "
+                    f"{classname!r} class."
+                    + (f"  Closest matches: {matches!r}" if matches else "")
```

Comment:
Since we've now got three copies of this, let's pull it out into a new `describe_close_matches()` function so that we can just do `+ describe_close_matches(classname, vars(module))` without the comprehension and call and ternary statement every time.

Reply:
sure